d3-dashboard/632.js

```diff
@@ -0,0 +1,198 @@
+    .attr('stroke', tumblrColor)
+    .style('cursor', 'pointer')
+
+  lineGraph.selectAll('instagram-circles')
+    .data(data)
+    .enter()
+    .append('circle')
+    .attr('cx', d => xScale(d.year))
+    .attr('cy', d => yScale(d.followers.instagram))
+    .attr('r', 6)
+    .attr('fill', 'white')
+    .attr('stroke', instagramColor)
+    .style('cursor', 'pointer')
+
+  const rightDashboard = d3.select('.dashboard')
+    .append('div');
+
+  const pieGraph = rightDashboard.append('svg')
+    .attr('width', 200)
+    .attr('height', 200)
+    .style('position', 'relative')
+    .style('left', '20px');
+
+  const pieArc = d3.arc()
+    .outerRadius(100)
+    .innerRadius(0);
+
+  const pieColors = d3.scaleOrdinal()  
+    .domain(data[index].followers)
+    .range([twitterColor, tumblrColor, instagramColor]);
+
+  const pie = d3.pie()
+    .value(d => d.value);
+    
+  const pieGraphData = pieGraph.selectAll('pieSlices')
+    .data(pie(d3.entries(data[index].followers)))
+    .enter()
+    .append('g')
+    .attr('transform', 'translate(100, 100)');
+
+  pieGraphData.append('path')
+    .attr('d', pieArc)
+    .attr('fill', d => pieColors(d.data.key))
+    .attr('stroke', 'white')
+    .attr('stroke-width', 2);
+
+  pieGraphData.selectAll('pieSliceText')
+    .data(pie(d3.entries(data[index].followers)))
+    .enter()
+    .append('text')
+    .text(d => `${Math.round(d.data.value/d3.sum(d3.values(data[index].followers))*100)}%`)
+    .attr('transform', d => `translate(${pieArc.centroid(d)})`)
+    .style('text-anchor', 'middle')
+    .style('font', '10px verdana');
+
+  const legend = rightDashboard.append('table')
+    .attr('width', 200)
+    .attr('height', 120)
+    .style('font', '12px verdana')
+    .style('position', 'relative')
+    .style('top', '30px');
+
+  const legendTitle = legend.append('thead')
+    .append('tr')
+    .append('th')
+    .text('2020 followers')
+    .attr('colspan', 3)
+    .style('position', 'relative')
+    .style('left', '20px');
+
+  const legendRows = legend.append('tbody')
+    .selectAll('tr')
+    .data(d3.entries(data[index].followers))
+    .enter()
+    .append('tr');
+
+  legendRows.append('td')  
+    .text(d => d.key)
+    .attr('align', 'right');
+
+  legendRows.append('td')
+    .attr('align', 'center')
+    .append('div')
+    .style('width', '16px')
+    .style('height', '16px')
+    .style('background-color', d => pieColors(d.key))
+
+  legendRows.append('td')
+    .text(d => d.value)
+    .attr('align', 'left');
+}
+
+drawDashboard(2020);
+
+/*
+  There's a problem, each time you call the function it adds more elements to the container. At the very top of the function, if you simply empty all the elements out of the container, it will redraw them where they need to be.
+
+  At the top of the function, use the `d3.select` function to select the `.dashboard` element and chain the `html` function to it with an empty string as it parameter. It should look like this:
+
+```
```

Comment:
This could be moved to the top of the function.

Reply:
Meaning, move the instructions to the top of the function?

Reply:
Yes, that's right. Generally we want the instructions to be above or below the line where learners will write their code like in the RPG game: https://github.com/freeCodeCamp/CurriculumExpansion/blob/master/basic-javascript-role-playing-game/index360.js

This is a minor issue, though, and I'd be happy to move the comments around in your next PR.